proposals/text/3665-multi-task-component.md

```diff
@@ -0,0 +1,309 @@
+                          prompt_text="Please give a sentiment for this context. Answer with positive, negative or neutral. Context: $documents; Answer:",
+                          prompt_params=["documents"]))
+      mtc.supported_task_names()
+
+      ----------------------------------------------------------------------------
+      ['question-answering',
+       'question-generation',
+       'conditioned-question-generation',
+       'summarization',
+       'question-answering-check',
+       'sentiment-analysis']
+  ```
+
+- Task parameters verification
+
+  All task parameters will be verified to match the task template and the corresponding object parameters will be
+  checked for type and value. For example:
+
+  ``` python
+      from haystack.nodes.llm import MultiTaskComponent
+      mtc = MultiTaskComponent(model_name_or_path="google/flan-t5-base")
+      mtc.prompt_for_task("question-generation", some_unknown_param=["Berlin is the capital of Germany."])
+
+      ----------------------------------------------------------------------------
+      ValueError                                Traceback (most recent call last)
+      <ipython-input-16-369cca52e960> in <module>
+            1 # tasks parameters are checked
+      ----> 2 sa(some_param=[Document("Berlin is the capital of Germany.")])
+
+      2 frames
+      /usr/local/lib/python3.8/dist-packages/haystack/nodes/llm/multi_task.py in __call__(self, *args, **kwargs)
+           34         if set(template_dict.keys()) != set(self.prompt_params):
+           35             available_params = set(list(template_dict.keys()) + list(set(kwargs.keys())))
+      ---> 36             raise ValueError(f"Expected prompt params {self.prompt_params} but got {list(available_params)}")
+           37
+           38         template_dict["prompt_template"] = self.prompt_text
+
+      ValueError: Expected prompt params ['documents'] but got ['some_unknown_param']
+  ```
+
+- MultiTaskPipelines
+
+  Even though we can use MultiTaskComponent directly its real power lies in using pipelines and Haystack. For example, we
+  can retrieve documents from the document store using the query and then inject the retrieved documents into documents
+  as a parameter to the selected MultiTaskComponent task. For example:
+
+  ``` python
+      from haystack.pipelines import MultiTaskPipeline
+      top_k = 3
+      query = "Who are the parents of Arya Stark?"
+      mtc = MultiTaskComponent(model_name_or_path="google/flan-t5-base")
+      pipe = MultiTaskPipeline(mtc, retriever)
+      output = pipe.run(query=query,
+                        params={"Retriever": {"top_k": top_k}},
+                        task="question-answering",
+                        questions=[query for n in range(0, top_k)],
+                        #documents parameter we need for this task will be automatically populated by the retriever
+                        )
+
+      output["results"]
```

Comment:
I'm missing the `retriever` in this snippet (and in other snippets below). Is it a regular Haystack retriever like `EmbeddingRetriever`?